api/src/Entity/Discussion.php

```diff
@@ -61,10 +72,39 @@
                 ],
             ],
         ],
-        summary: 'Get number of message not read',
-        description: 'Get number of message not read for all discussions',
+        summary: 'Get number of message not read for a repairer',
+        description: 'Get number of message not read for all discussions of a repairer',
     ),
     paginationEnabled: false,
+    security: 'is_granted("IS_AUTHENTICATED_FULLY") and user.isAssociatedWithRepairer(repairer_id)',
+)]
+#[GetCollection(
+    uriTemplate: '/customers/messages_unread',
```

Comment:
The job is done and it works so I'm okay :), but instead of using this operation on a Discussion resource and using a controller for returning a response, a better way would be to use another resource, let's call it `MessageUnread` for example (in `api/src/Message/Resource/ or api/src/Discussion/Resource`) and add it a Get operation with a provider like:

```
#[Get(
    '/customers/messages_unread',
    security: 'is_granted("IS_AUTHENTICATED_FULLY")',
    provider: NumberOfMessageNotReadForCustomerProvider::class,
)]
final class MessageUnread
{
    public function __construct(
        public int $count,
    ) {
    }
}

// provider
readonly final class NumberOfMessageNotReadForCustomerProvider implements ProviderInterface
{
    public function __construct(
        private DiscussionMessageRepository $discussionMessageRepository,
        private Security $security,
    ) {
    }

    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
    {
        /** @var User|null $user */
        $user = $this->security->getUser();
        $notRead = $user ? $this->discussionMessageRepository->getNumberOfMessageNotReadForCustomer(user: $user) : 0;

        return new MessageUnread($notRead);
    }
}

// response will be
{
    "@context": "/contexts/MessageUnread",
    "@id": "/customers/messages_unread",
    "@type": "MessageUnread",
    "count": 10
}

```